### ISSUE TITLE
Remove left icon space on MenuItem

### ISSUE THREAD

**Issue by basix86** [JFormDesigner/FlatLaf#328]
Suppose to have a JMenu with only JMenuItems without icons.
We wold avoid the extra left space reserved for the icons. The same behavior is present in BasicMenuItemUI

![menuItemUI](https://user-images.githubusercontent.com/17149962/118819531-b8907b80-b8b5-11eb-994b-9ac1f9429b88.png)

**Comment by DevCharly** [comment]
I agree that the menu item layout should be more flexible. Currently it always reserves extra space for an icon.

You can set `MenuItem.minimumIconSize` to `0,0` to reduce the space reserved for icons,
but this looks bad if some menu items have icons and some not.
Use following before creating menu items:
~~~java
UIManager.put( "MenuItem.minimumIconSize", new Dimension() );
~~~

Here is a trick to create a single popup menu without extra space reserved for icons:
~~~java
UIManager.put( "MenuItem.minimumIconSize", new Dimension() );

// create popup menu items here

UIManager.put( "MenuItem.minimumIconSize", null );
~~~
